hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightSearchEndpoint.java

```diff
@@ -0,0 +1,383 @@
+import static org.apache.hadoop.ozone.recon.ReconResponseUtils.createInternalServerErrorResponse;
+import static org.apache.hadoop.ozone.recon.ReconUtils.constructObjectPathWithPrefix;
+import static org.apache.hadoop.ozone.recon.ReconUtils.validateNames;
+import static org.apache.hadoop.ozone.recon.api.handlers.BucketHandler.getBucketHandler;
+import static org.apache.hadoop.ozone.recon.api.handlers.EntityHandler.normalizePath;
+import static org.apache.hadoop.ozone.recon.api.handlers.EntityHandler.parseRequestPath;
+
+/**
+ * REST endpoint for search implementation in OM DB Insight.
+ */
+@Path("/keys")
+@Produces(MediaType.APPLICATION_JSON)
+@AdminOnly
+public class OMDBInsightSearchEndpoint {
+
+  private OzoneStorageContainerManager reconSCM;
+  private final ReconOMMetadataManager omMetadataManager;
+  private static final Logger LOG =
+      LoggerFactory.getLogger(OMDBInsightSearchEndpoint.class);
+  private ReconNamespaceSummaryManagerImpl reconNamespaceSummaryManager;
+
+
+  @Inject
+  public OMDBInsightSearchEndpoint(OzoneStorageContainerManager reconSCM,
+                                   ReconOMMetadataManager omMetadataManager,
+                                   ReconNamespaceSummaryManagerImpl reconNamespaceSummaryManager) {
+    this.reconSCM = reconSCM;
+    this.omMetadataManager = omMetadataManager;
+    this.reconNamespaceSummaryManager = reconNamespaceSummaryManager;
+  }
+
+
+  /**
+   * Performs a search for open keys in the Ozone Manager (OM) database using a specified search prefix.
+   * This endpoint searches across both File System Optimized (FSO) and Object Store (non-FSO) layouts,
+   * compiling a list of keys that match the given prefix along with their data sizes.
+   * <p>
+   * The search prefix must start from the bucket level ('/volumeName/bucketName/') or any specific directory
+   * or key level (e.g., '/volA/bucketA/dir1' for everything under 'dir1' inside 'bucketA' of 'volA').
+   * The search operation matches the prefix against the start of keys' names within the OM DB.
+   * <p>
+   * Example Usage:
+   * 1. A startPrefix of "/volA/bucketA/" retrieves every key under bucket 'bucketA' in volume 'volA'.
+   * 2. Specifying "/volA/bucketA/dir1" focuses the search within 'dir1' inside 'bucketA' of 'volA'.
+   *
+   * @param startPrefix The prefix for searching keys, starting from the bucket level or any specific path.
+   * @param limit       Limits the number of returned keys.
+   * @return A KeyInsightInfoResponse, containing matching keys and their data sizes.
+   * @throws IOException On failure to access the OM database or process the operation.
+   */
+  @GET
+  @Path("/open/search")
+  public Response searchOpenKeys(
+      @DefaultValue(DEFAULT_START_PREFIX) @QueryParam("startPrefix")
+      String startPrefix,
+      @DefaultValue(RECON_OPEN_KEY_DEFAULT_SEARCH_LIMIT) @QueryParam("limit")
+      int limit) throws IOException {
```

Comment:
Thanks @ArafatKhan2198 for updating the patch, however client still cannot query the next page because API is not accepting the prevKey as param, Pls update the API and add test cases to traverse multiple pages and add assertions per page.

Reply:
Thanks for the mention @devmadhuu Its been handled 
cc: @dombizita